src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1940,15 +1946,15 @@ private Task WriteRowSourceToServerAsync(int columnCount, CancellationToken ctok
                         {
                             AsyncHelper.ContinueTaskWithState(writeTask, tcs,
                                 state: tcs,
-                                onSuccess: state => ((TaskCompletionSource<object>)state).SetResult(null)
+                                onSuccess: static (object state) => ((TaskCompletionSource<object>)state).SetResult(null)
                             );
                         }
                     }, ctoken); // We do not need to propagate exception, etc, from reconnect task, we just need to wait for it to finish.
                     return tcs.Task;
                 }
                 else
                 {
-                    AsyncHelper.WaitForCompletion(reconnectTask, BulkCopyTimeout, () => { throw SQL.CR_ReconnectTimeout(); }, rethrowExceptions: false);
+                    AsyncHelper.WaitForCompletion(reconnectTask, BulkCopyTimeout, static () => throw SQL.CR_ReconnectTimeout(), rethrowExceptions: false);
```

Comment:
Yes, that is the purpose. There is also a useful side effect of the way that the compiler implements it because it's a static delegate means that the delegate can be cached. This [sharplab example](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oYaY+WRiqQEIAUAlA03QSZCAggGcAngDswACRgAbAA4woAOgDqAQwCWAFwBi0ShAC2i+TF3aIkzpICu8+QBpSABlcYcSUj1IBeAD5SXQALKAgAd1IAZQBFABlVSgAlAH0UmEhJSSzdABVtExgIe10eV1gwiMiAUQQwGEUrG1EQUgAzTXlRGG4Abj5GAF9CUYJCEnIsUjEpWQVlKEIBISmvH1ZSLT1DKGMzCxbbCGAAKzzSWGzcsF0MJGdtSV0QopKyzxwZzTvrSVcwAgEHkV0s4Si9UazX+om4tHG4yEkxYM3iCRWQ2Y5G8pChTWOVHSmRueUKxVK5XhNAwAHZSLloviYTYeP1EUA=) shows that you get a hidden static class generated and that the action delegate is created the first time it is needed and re-used every time it is called after that. So we get lazy single allocation for free with a nice syntax.

```csharp
    public void B()
    {
       AsyncHelper.WaitForCompletion(null, 0, static () => throw SQL.CR_ReconnectTimeout(), rethrowExceptions: false);
    }
```

is lowered to:

```csharp
    [Serializable]
    [CompilerGenerated]
    private sealed class <>c
    {
        public static readonly <>c <>9 = new <>c();

        public static Action <>9__0_0;

        internal void <B>b__0_0()
        {
            throw SQL.CR_ReconnectTimeout();
        }
    }

    public void B()
    {
        AsyncHelper.WaitForCompletion(null, 0, <>c.<>9__0_0 ?? (<>c.<>9__0_0 = new Action(<>c.<>9.<B>b__0_0)), false);
    }
```